src/interpreter/vm/DurianObject.cpp

```diff
@@ -12,8 +12,8 @@ DurianObject::DurianObject(double dval) :
     value.dval = dval;
 }
 
-DurianObject::DurianObject(int64_t ival) :
-    type(DurianType::Integer) {
+DurianObject::DurianObject(int64_t ival, DurianType type = DurianType::Integer) :
```

Comment:
Perhaps we should have it check for `type` and crash if there's something unexpected?

Reply:
And by "crash" I mean return throw an exception or something similarly reasonable... this isn't Erlang 😛